### PR TITLE
Fix staticchecks in vendor/k8s.io/client-go

### PR DESCRIPTION
**Pull request by Git-Jiro**
**What type of PR is this?**
/kind cleanup

**Which issue(s) this PR fixes**:
Part of #92402
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by fedebongio** [comment]
/assign @caesarxuchao 
thank you!

File: staging/src/k8s.io/client-go/restmapper/category_expansion.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func NewDiscoveryCategoryExpander(client discovery.DiscoveryInterface) CategoryE
 // Expand fulfills CategoryExpander
 func (e discoveryCategoryExpander) Expand(category string) ([]schema.GroupResource, bool) {
 	// Get all supported resources for groups and versions from server, if no resource found, fallback anyway.
-	apiResourceLists, _ := e.discoveryClient.ServerResources()
+	_, apiResourceLists, _ := e.discoveryClient.ServerGroupsAndResources()
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/restmapper/category_expansion.go:61]: Why is this change needed?

Comment (Git-Jiro) [reply staging/src/k8s.io/client-go/restmapper/category_expansion.go:61]: Because the static check said that ServerResources is a deprecated function

**Comment by Git-Jiro** [comment]
@sttts please take a quick look at hack/.staticcheck_failures

File: staging/src/k8s.io/client-go/dynamic/fake/simple.go [hunk 2]
```diff
@@ -116,6 +119,9 @@ func (c *dynamicResourceClient) Create(ctx context.Context, obj *unstructured.Un
 		name := accessor.GetName()
 		uncastRet, err = c.client.Fake.
 			Invokes(testing.NewCreateSubresourceAction(c.resource, name, strings.Join(subresources, "/"), c.namespace, obj), obj)
+		if err != nil {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/dynamic/fake/simple.go:122]: There is a check on err after the switch, why isn't that enough?

Also why some switch cases don't have this error check?

File: staging/src/k8s.io/client-go/restmapper/category_expansion.go (hunk 1 above)
Comment (caesarxuchao) [reply staging/src/k8s.io/client-go/restmapper/category_expansion.go:61]: Got it. Thanks for the explanation.

File: staging/src/k8s.io/client-go/metadata/fake/simple.go [hunk 3]
```diff
@@ -121,6 +121,9 @@ func (c *metadataResourceClient) CreateFake(obj *metav1.PartialObjectMetadata, o
 		name := accessor.GetName()
 		uncastRet, err = c.client.Fake.
 			Invokes(testing.NewRootCreateSubresourceAction(c.resource, name, strings.Join(subresources, "/"), obj), obj)
+		if err != nil {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/metadata/fake/simple.go:124]: same question here.

File: staging/src/k8s.io/client-go/dynamic/fake/simple.go (hunk 2 above)
Comment (Git-Jiro) [reply staging/src/k8s.io/client-go/dynamic/fake/simple.go:122]: Now that you are mentioning it ... yes, I agree.
Those additional if statements do not make any sense at all!

Comment (Git-Jiro) [reply staging/src/k8s.io/client-go/dynamic/fake/simple.go:122]: Not quite! The static check noticed that err was shadowed and therefore the err was not evaluated.

Comment (caesarxuchao) [reply staging/src/k8s.io/client-go/dynamic/fake/simple.go]: Good catch!

**Comment by caesarxuchao** [comment]
If you see this message before it's merged, can you squash the commits?

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
